### PR TITLE
Json rpc set host

### PR DESCRIPTION
**Pull request by diega**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 18 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [RskSystemProperties.java#L176](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L176): Make this IP "0.0.0.0" address configurable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1313)
1. ![MAJOR][MAJOR] [Start.java#L65](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/Start.java#L65): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L83](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/Start.java#L83): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L72](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L72): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L312](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L312): Method co.rsk.config.RskSystemProperties.getRpcModules() does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L346](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L346): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L347](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L347): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L349](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L349): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L357](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L357): java/io/FileOutputStream.<init>(Ljava/lang/String;)V writes to a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_OUT)
1. ![MAJOR][MAJOR] [JsonRpcNettyServer.java#L55](https://github.com/rsksmart/rskj/blob/18ade1ef47edb4cbb4b563460e39ec8c437def56/rskj-core/src/main/java/org/ethereum/rpc/JsonRpcNettyServer.java#L55): Reduce this anonymous class number of lines from 21 to at most 20, or make it a named class. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1188)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 6 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Start.java#L65](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/Start.java#L65): Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [Start.java#L83](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/Start.java#L83): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L74](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L74): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L322](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L322): Method co.rsk.config.RskSystemProperties.getRpcModules() does not presize the allocation of a collection [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APSC_PRESIZE_COLLECTIONS)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L356](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L356): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L357](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L357): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L359](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L359): java/nio/file/Paths.get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path; reads a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_IN)
1. ![MAJOR][MAJOR] [RskSystemProperties.java#L367](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java#L367): java/io/FileOutputStream.<init>(Ljava/lang/String;)V writes to a file whose location might be specified by user input [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3APATH_TRAVERSAL_OUT)
1. ![MAJOR][MAJOR] [JsonRpcNettyServer.java#L57](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/org/ethereum/rpc/JsonRpcNettyServer.java#L57): Reduce this anonymous class number of lines from 21 to at most 20, or make it a named class. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1188)
1. ![MINOR][MINOR] [Start.java#L65](https://github.com/rsksmart/rskj/blob/f303397729fac8aca7692eae438c7885aa71bba0/rskj-core/src/main/java/co/rsk/Start.java#L65): Move this constructor to comply with Java Code Conventions. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1213)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/rpc/JsonRpcNettyServer.java [hunk 1]
```diff
@@ -65,7 +75,7 @@ protected void initChannel(SocketChannel ch) throws Exception {
                     p.addLast(jsonRpcWeb3ServerHandler);
                 }
             });
-        b.bind(port).sync();
+        b.bind(host, port).sync();
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/rpc/JsonRpcNettyServer.java]: Note that this line is the one where we stop binding to `0.0.0.0` and bind to a specific host instead. By default, we'll bind to the loopback address now :clap:.

File: rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java [hunk 2]
```diff
@@ -171,6 +173,19 @@ public int rpcPort() {
...
 
+    public InetAddress rpcHost() {
+        if (!configFromFiles.hasPath("rpc.host")) {
+            return InetAddress.getLoopbackAddress();
+        }
+        String host = configFromFiles.getString("rpc.host");
+        try {
+            return InetAddress.getByName(host);
+        } catch (UnknownHostException e) {
+            logger.warn("Unable to bind to {}. Using loopback instead", e);
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/config/RskSystemProperties.java]: Missing parameter.

**Pull request merged**